tracing-appender/src/lib.rs

```diff
@@ -19,23 +21,27 @@ mod worker;
 /// occur if logs are consistently produced faster than the worker thread can
 /// output them. The queue capacity and behavior when full (i.e., whether to
 /// drop logs or to exert backpressure to slow down senders) can be configured
-/// using [`NonBlockingBuilder::default()`]. This function simply returns the default
-/// configuration &mdash; it is equivalent to
+/// using [`NonBlockingBuilder::default()`][builder].
+/// This function simply returns the default configuration &mdash; it is equivalent to
 ///
 /// ```rust
 /// # use tracing_appender::non_blocking::{NonBlocking, WorkerGuard};
 /// # fn doc() -> (NonBlocking, WorkerGuard) {
 /// tracing_appender::non_blocking::NonBlocking::new(std::io::stdout())
 /// # }
 /// ```
-/// [`NonBlocking::builder()`]: /non_blocking/struct.NonBlocking.html#method.builder
+/// [builder]: non_blocking/struct.NonBlockingBuilder.html#method.default
+///
+/// <br/> This function returns a tuple containing a `NonBlocking` struct and a `WorkerGuard`. The
+/// NonBlocking` struct implements [`MakeWriter`], while the `WorkerGuard` is a drop guard which
+/// is responsible for ensuring any buffered logs are flushed when the program terminates.
 ///
 /// Note that the `WorkerGuard` returned by `non_blocking` _must_ be assigned to a binding that
-/// is not `_`, as `_` will result in the `WorkerGuard` being dropped immediately.
+/// is not `_`, as `_` will result in the `WorkerGuard` being dropped immediately. It should also
+/// not be dropped accidently if you want to ensure logs are flushed during panics.
```

Comment:
```suggestion
/// is not `_`, as `_` will result in the `WorkerGuard` being dropped immediately.
/// Unintentional drops of `WorkerGuard` remove the guarantee that logs will be flushed
/// during a program's termination, in a panic or otherwise.
```